TASVideos.WikiEngine/Node.cs

```diff
@@ -22,6 +22,8 @@ public interface INode
 	/// </summary>
 	Task WriteTextAsync(TextWriter writer, WriterContext ctx);
 
+	Task WriteMetaDescriptionAsync(StringBuilder sb, WriterContext ctx);
```

Comment:
Could these take a `TextWriter` instead of a `StringBuilder` for consistency with other node implementations?  This shouldn't require significant changes to the caller as we can use `StringWriter` there as a concrete `TextWriter` implementation.

Reply:
No, because I need the length of the currently written text. TextWriters don't have that ability.

Reply:
Could the truncation be the responsibility of a custom `TextWriter` implementation?

Reply:
Truncating would work, yes, but I'm using the length for deciding whether to go through more nodes, to avoid having to iterate the entire node tree once we already have enough.

Reply:
> to avoid having to iterate the entire node tree once we already have enough.

I think that's a good idea, but I don't see it happening in most places; I'll open separate comments there.

Reply:
Resolving this as I think it makes sense to stick with the stringbuilder.